libbpf-rs/src/object.rs

```diff
@@ -334,6 +353,58 @@ impl Object {
         self.map(name).unwrap().unwrap()
     }
 
+
+    pub fn get_maps(&mut self) -> Result<&HashMap<String, Map>> {
+
+        // if populated already return the populated hashmap
+        if !self.maps.is_empty() {
+            Ok(&self.maps)
+
+        } else {
+     
+            let mut maps: HashMap<String, Map> = HashMap::new();
+            let mut map: *mut libbpf_sys::bpf_map = std::ptr::null_mut();
+
+            loop {
+                // while there's another map to grab
+                if let Some(next_map_ptr) = object_map_next(self.ptr, map).unwrap() {
+
+                    // get map name
+                    let mname = unsafe { libbpf_sys::bpf_map__name(next_map_ptr) };
+                    let err = unsafe { libbpf_sys::libbpf_get_error(mname as *const _) };
+                    if err != 0 {
+                        std::process::exit(err as i32);
```

Comment:
As a library, we should never cause applications to exit. This should be

```suggestion
                        return Err(Error::System(err as i32));

Reply:
This is my fault for not reviewing my code as much as I should've before making a PR.